substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrFeature.java

```diff
@@ -112,6 +116,46 @@ public void beforeAnalysis(Feature.BeforeAnalysisAccess access) {
         runtime.addShutdownHook(manager::teardown);
     }
 
+    @SuppressWarnings(value = "unchecked")
+    private static void addEventClassToRuntime(JfrRuntimeAccess jfrRuntime, Class<?> cls) {
+        jfrRuntime.addEventClass((Class<? extends Event>) cls);
+    }
+
+    public void afterAnalysis(AfterAnalysisAccess access) {
+//        if (!JfrAvailability.withJfr) {
```

Comment:
The Feature now has some overrides that should mean this isn't necessary anymore:

```
    @Override
    public boolean isInConfiguration(IsInConfigurationAccess access) {
        return JfrEnabled.get();
    }

    @Override
    public List<Class<? extends Feature>> getRequiredFeatures() {
        return Collections.singletonList(ThreadListenerFeature.class);
    }
```

My understanding is the feature won't be included unless -H+AllowVMInspection is supplied and the jdk is >= 11.

Reply:
Er, to conclude, these if statements for availability can be dropped.